File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-0.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int one = 0;
        int two = 0;
        for (int i = cost.length - 1; i >= 0; i--) {
            int c = cost[i] + Math.min(one, two);
            two = one;
            one = c;
        }
        return Math.min(one, two);
    }
}

File: Data Structures & Algorithms/decode-ways/submission-0.py
class Solution:
    def numDecodings(self, s: str) -> int:
        dp = [0 for _ in range(len(s) + 1)]
        # num ways for s[:i]

        if s[0] == "0": return 0
        dp[0] = 1
        dp[1] = 1

        for i in range(2, len(s) + 1):
            one = int(s[i - 1])
            two = int(s[i - 2 : i])

            if 1 <= one <= 9:
                dp[i] += dp[i - 1]
            
            if 10 <= two <= 26:
                dp[i] += dp[i - 2]

        return dp[-1]